toolchain/private/defs.bzl

```diff
@@ -199,3 +200,25 @@ def _target_linux_musl(gocpu, zigcpu):
         ld_zig_subcmd = "ld.lld",
         artifact_name_patterns = [],
     )
+
+def _target_wasm():
+    return struct(
+        gotarget = "wasip1_wasm",
+        zigtarget = "wasm32-wasi-musl",
+        includes = [
+          "libc/include/wasm-wasi-musl",
+          "libc/wasi",
+        ] + _INCLUDE_TAIL,
+        linkopts = [],
+        dynamic_library_linkopts = [],
+        supports_dynamic_linker = True,
```

Comment:
TBH I believe it does work, but I have not tested it - I generally statically link, since wasm binaries have to be self contained at the end of the day. I do believe there is some support for shared libraries, but I don't know the details. I'll update this to just remove it for now.